libs/langchain/langchain/vectorstores/redis/schema.py

```diff
@@ -0,0 +1,158 @@
+from typing import TYPE_CHECKING, List, Optional, Union
+
+from pydantic import BaseModel, Field, validator
+from redis.commands.search.field import (
+    GeoField,
+    NumericField,
+    TagField,
+    TextField,
+    VectorField,
+)
+from typing_extensions import Literal
+
+if TYPE_CHECKING:
+    from redis.commands.search.field import Field as RedisField
+
+
+class BaseField(BaseModel):
+    name: str = Field(...)
+    sortable: Optional[bool] = False
+
+
+class TextFieldSchema(BaseField):
+    weight: Optional[float] = 1
```

Comment:
out of curiosity why are all of these Optional if they have non-null defaults (other than phoenetic_matcher)? is that based on TextField